newIDE/app/src/MainFrame/index.js

```diff
@@ -841,107 +841,112 @@ const MainFrame = (props: Props) => {
+        let content;
+        try {
+          const result = await onOpen(
+            autoSavefileMetadata,
+            setLoaderModalProgress
+          );
+          content = result.content;
+        } catch (error) {
           // onOpen failed, tried to find again an autosave
-          return checkForAutosaveAfterFailure().then(fileMetadata => {
-            if (fileMetadata) {
-              return onOpen(fileMetadata);
-            }
-
-            throw err;
-          });
-        })
-        .then(({ content }) => {
-          if (!verifyProjectContent(i18n, content)) {
-            // The content is not recognized and the user was warned. Abort the opening.
-            setIsLoadingProject(false);
-            setLoaderModalProgress(null, null);
-            return;
+          const autoSaveAfterFailurefileMetadata = await checkForAutosaveAfterFailure();
+          if (autoSaveAfterFailurefileMetadata) {
+            const result = await onOpen(autoSaveAfterFailurefileMetadata);
+            content = result.content;
           }
+        }
+        if (!verifyProjectContent(i18n, content)) {
+          // The content is not recognized and the user was warned. Abort the opening.
+          setIsLoadingProject(false);
+          setLoaderModalProgress(null, null);
+          return;
+        }
+
+        const serializedProject = gd.Serializer.fromJSObject(content);
 
-          const serializedProject = gd.Serializer.fromJSObject(content);
-          return loadFromSerializedProject(
+        try {
+          const state = loadFromSerializedProject(
             serializedProject,
             // Note that fileMetadata is the original, unchanged one, even if we're loading
             // an autosave. If we're for some reason loading an autosave, we still consider
             // that we're opening the file that was originally requested by the user.
             fileMetadata
-          ).then(
-            state => {
-              serializedProject.delete();
-              return Promise.resolve(state);
-            },
-            err => {
-              serializedProject.delete();
-              throw err;
-            }
           );
-        })
-        .catch(error => {
-          const errorMessage = getOpenErrorMessage
-            ? getOpenErrorMessage(error)
-            : t`Check that the path/URL is correct, that you selected a file that is a game file created with GDevelop and that is was not removed.`;
-          showErrorBox({
-            message: [
-              i18n._(t`Unable to open the project.`),
-              i18n._(errorMessage),
-            ].join('\n'),
-            errorId: 'project-open-error',
-            rawError: error,
-          });
-          setIsLoadingProject(false);
-          setLoaderModalProgress(null, null);
-          return Promise.reject(error);
+          serializedProject.delete();
+          return Promise.resolve(state);
```

Comment:
nit: might not need the Promise.resolve?

Reply:
oopsie